client/blocks/image-editor/index.jsx

```diff
@@ -190,22 +205,46 @@ const ImageEditor = React.createClass( {
 		return imageProperties;
 	},
 
-	onLoadCanvasError() {
-		const { translate } = this.props;
+	showNotice( noticeText, noticeStatus = 'is-info' ) {
+		// check noticeText in case translate() returns an unexpected value
+		if ( typeof noticeText !== 'string' ) {
+			noticeText = null;
+		}
+		this.setState( {
+			noticeText,
+			noticeStatus
+		} );
+	},
 
+	onDismissNotice() {
 		this.setState( {
-			canvasError: translate( 'We are unable to edit this image.' )
+			noticeText: null
 		} );
 	},
 
-	renderError() {
+	renderNotice() {
+		if ( ! this.state.noticeText ) {
+			return null;
+		}
+		const onDismissClick = this.state.noticeStatus === 'is-error'
+			? this.onCancel
+			: this.onDismissNotice;
```

Comment:
Naming wise let's aim to be descriptive of intent. Since we're clearing state, this might be `this.clearNoticeState` instead of `this.onDismissNotice`